tests/test_node.py

```diff
@@ -676,3 +678,383 @@ def render(node: BaseNode, context: Context, **kwargs) -> str:
             template2.render(Context({}))
 
         render._node.unregister(component_tags.register)  # type: ignore[attr-defined]
+
+
+def force_signature_validation(fn):
+    """
+    Creates a proxy around a function that makes __code__ inaccessible,
+    forcing the use of signature-based validation.
+    """
+
+    class SignatureOnlyFunction:
+        def __init__(self, fn):
+            self.__wrapped__ = fn
+            self.__signature__ = inspect.signature(fn)
+
+        def __call__(self, *args, **kwargs):
+            return self.__wrapped__(*args, **kwargs)
+
+        def __getattr__(self, name):
+            # Return None for __code__ to force signature-based validation
+            if name == "__code__":
+                return None
+            # For all other attributes, delegate to the wrapped function
+            return getattr(self.__wrapped__, name)
+
+    return SignatureOnlyFunction(fn)
+
+
+class SignatureBasedValidationTests(BaseTestCase):
```

Comment:
Added tests for the signature-based validation, as the `__code__` base one is now the default - or at least I think it's fairly safe to assume that any one of us who will run tests will be using CPython / Pypy